v2/api/network/v1api20180501/dns_zone_types_gen.go

```diff
@@ -0,0 +1,1361 @@
+func (zone *DnsZone) validateWriteOnceProperties(old runtime.Object) error {
+	oldObj, ok := old.(*DnsZone)
+	if !ok {
+		return nil
+	}
+
+	return genruntime.ValidateWriteOnceProperties(oldObj, zone)
+}
+
+// AssignProperties_From_DnsZone populates our DnsZone from the provided source DnsZone
+func (zone *DnsZone) AssignProperties_From_DnsZone(source *v1api20180501s.DnsZone) error {
+
+	// ObjectMeta
+	zone.ObjectMeta = *source.ObjectMeta.DeepCopy()
+
+	// Spec
+	var spec DnsZone_Spec
+	err := spec.AssignProperties_From_DnsZone_Spec(&source.Spec)
+	if err != nil {
+		return errors.Wrap(err, "calling AssignProperties_From_DnsZone_Spec() to populate field Spec")
+	}
+	zone.Spec = spec
+
+	// Status
+	var status DnsZone_STATUS
+	err = status.AssignProperties_From_DnsZone_STATUS(&source.Status)
+	if err != nil {
+		return errors.Wrap(err, "calling AssignProperties_From_DnsZone_STATUS() to populate field Status")
+	}
+	zone.Status = status
+
+	// No error
+	return nil
+}
+
+// AssignProperties_To_DnsZone populates the provided destination DnsZone from our DnsZone
+func (zone *DnsZone) AssignProperties_To_DnsZone(destination *v1api20180501s.DnsZone) error {
+
+	// ObjectMeta
+	destination.ObjectMeta = *zone.ObjectMeta.DeepCopy()
+
+	// Spec
+	var spec v1api20180501s.DnsZone_Spec
+	err := zone.Spec.AssignProperties_To_DnsZone_Spec(&spec)
+	if err != nil {
+		return errors.Wrap(err, "calling AssignProperties_To_DnsZone_Spec() to populate field Spec")
+	}
+	destination.Spec = spec
+
+	// Status
+	var status v1api20180501s.DnsZone_STATUS
+	err = zone.Status.AssignProperties_To_DnsZone_STATUS(&status)
+	if err != nil {
+		return errors.Wrap(err, "calling AssignProperties_To_DnsZone_STATUS() to populate field Status")
+	}
+	destination.Status = status
+
+	// No error
+	return nil
+}
+
+// OriginalGVK returns a GroupValueKind for the original API version used to create the resource
+func (zone *DnsZone) OriginalGVK() *schema.GroupVersionKind {
+	return &schema.GroupVersionKind{
+		Group:   GroupVersion.Group,
+		Version: zone.Spec.OriginalVersion(),
+		Kind:    "DnsZone",
+	}
+}
+
+// +kubebuilder:object:root=true
+// Generator information:
+// - Generated from: /dns/resource-manager/Microsoft.Network/stable/2018-05-01/dns.json
+// - ARM URI: /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/dnsZones/{zoneName}
+type DnsZoneList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []DnsZone `json:"items"`
+}
+
+// +kubebuilder:validation:Enum={"2018-05-01"}
+type APIVersion string
+
+const APIVersion_Value = APIVersion("2018-05-01")
+
+type DnsZone_Spec struct {
+	// AzureName: The name of the resource in Azure. This is often the same as the name of the resource in Kubernetes but it
+	// doesn't have to be.
+	AzureName string `json:"azureName,omitempty"`
+
+	// Etag: The etag of the zone.
+	Etag *string `json:"etag,omitempty"`
```

Comment:
I am pretty sure that this field should be readonkly? Maybe fix it in our config, and/or in Swagger upstream?

Reply:
Did not seem like a read-only field to me - https://learn.microsoft.com/en-us/azure/dns/dns-zones-records#etags

Reply:
Are you sure? They say this:
"At the level of the Azure DNS REST API, Etags are specified using HTTP headers. Their behavior is given in the following table:"

So they are basically saying what you do is, you do a GET, you take the value of the Etag returned by the server, and you put it as the HTTP header `if-none-match`/`if-match`, etc.

AFAIK the value in the body is always only read from the server. The instruction to the server comes in the form of an HTTP header `if-none-match`/`if-match`. Because Etags are only ever generated by the server, it doesn't make any sense to allow the clients to set them - what would they set them to? They don't understand how to make new ones.

At least, that's how all the ETag's I've ever used work.

Reply:
Maybe try setting it to something and see what the server does? They might just ignore it in which case that doesn't really tell us anything, but they might reject.

Reply:
I'm 99% sure that this should be readonly though